src/index.js

```diff
@@ -1,6 +1,29 @@
-export default () => {
-  const greeting = 'Hello from new repository';
-  console.log(greeting);
+import axios from 'axios';
+import fsp from 'fs/promises';
+import path from 'path';
 
-  return greeting;
+const fetchPageData = (url) => axios.get(url)
+  .then((response) => response.data)
+  .catch((error) => {
+    throw error;
+  });
+
+const getFileNameFromUrl = (url) => {
+  const [, urlWithoutScheme] = url.split('://');
+  const filename = urlWithoutScheme.split(/[^a-zA-Z0-9]/).join('-').concat('.html');
```

Comment:
```suggestion
const filename = urlWithoutScheme.replace(/[^a-zA-Z0-9]/g, '-').concat('.html')
```

такой вариант так же проще